### PR TITLE
specify which Messages to delete with text

### PR DESCRIPTION
**Pull request by vladak**
Trying to use the `opengrok-mirror` handling of disabled projects (#2973) I discovered this gets blocked by cleanup handling in `opengrok-sync` configuration:
```yml
commands:
- command:
  - http://localhost:8080/source/api/v1/messages
  - POST
  - cssClass: info
    duration: PT1H
    tags: ['%PROJECT%']
    text: resync + reindex in progress

... opengrok-mirror with disabled project handling emits the message

cleanup:
  - command: ['http://localhost:8080/source/api/v1/messages?tag=%PROJECT%', DELETE, '']
```
To make this possible, I extended the `DELETE` handling in `MessagesController` to accept entity body (this is possible due to HTTP 1.1 spec RFC 7231 as explained on https://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request?rq=1) and match also on that.

Tested with Tomcat 8.5 and curl (both allow DELETE requests with body out of the box).

### CODE REVIEW

File: opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/MessagesControllerTest.java [hunk 1]
```diff
@@ -72,6 +84,67 @@ protected Application configure() {
         return new ResourceConfig(MessagesController.class);
     }
 
+    // Allow entity body for DELETE method on the client side.
+    @Override
+    protected void configureClient(ClientConfig config) {
+        config.property(ClientProperties.SUPPRESS_HTTP_COMPLIANCE_VALIDATION, true);
```
Comment (vladak) [inline opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/MessagesControllerTest.java]: per https://stackoverflow.com/questions/36310863/jersey-test-http-delete-method-with-json-request and https://stackoverflow.com/questions/25229880/how-to-send-enclose-data-in-delete-request-in-jersey-client

File: opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/MessagesControllerTest.java [hunk 2]
```diff
@@ -72,6 +84,67 @@ protected Application configure() {
...
+        }
+
+        private class GrizzlyTestContainer implements TestContainer {
+            private URI baseUri;
+            private final HttpServer server;
+
+            private GrizzlyTestContainer(URI baseUri, DeploymentContext context) {
+                this.baseUri = UriBuilder.fromUri(baseUri).path(context.getContextPath()).build(new Object[0]);
+                this.server = GrizzlyHttpServerFactory.createHttpServer(this.baseUri, context.getResourceConfig(), false);
+                this.server.getServerConfiguration().setAllowPayloadForUndefinedHttpMethods(true);
```
Comment (vladak) [inline opengrok-web/src/test/java/org/opengrok/web/api/v1/controller/MessagesControllerTest.java]: per https://stackoverflow.com/questions/49976638/allow-delete-requests-with-message-body-using-grizzly-servlet-container#49984647

File: opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java [hunk 3]
```diff
@@ -1935,11 +1935,11 @@ public void addMessage(final Message message) {
 
     /**
      * Remove all messages containing at least one of the tags.
-     *
+     * @param text message text (can be null, empty)
```
Comment (idodeclare) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java]: Nit: out of order. Otherwise LGTM

**Comment by vladak** [comment]
Thanks Chris !

**Comment by vladak** [comment]
The changes in f9e1941 and 6f98618 were result of internal review done by eric.saint.etienne@oracle.com.

**Pull request merged**